var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -364,7 +364,8 @@ def cmake_args(self):
                 [
                     "-DCUDA_TOOLKIT_ROOT_DIR:PATH=" + spec["cuda"].prefix,
                     "-DLIBOMPTARGET_NVPTX_COMPUTE_CAPABILITIES={0}".format(
-                        ",".join(spec.variants["cuda_arch"].value)
+                        ",".join(spec.variants["cuda_arch"].value),
+                    "-DLIBOMPTARGET_ENABLE_DEBUG=ON"
```

Comment:
Should this be turned on depending on the value of `build_type`? (Debug and RelWithDebInfo come to mind as canidates)?

Reply:
Well, please don't be confused with the LLVM build type, and I don't think this (-DLIBOMPTARGET_ENABLE_DEBUG setting to ON) slows down the `spack install llvm+cuda cuda_arch=70` process significantly, thus I hardcoded.

Reply:
@ikitayama are you sure you cannot control this already via +omp_debug?

Reply:
User can turn it on or off at run time by setting LIBOMPTARGET_DEBUG appropriately.

Reply:
I am trying to move my manual build of LLVM with OpenMP offloading support to a clean Spack build. You guys review, and suggestions are appreciated.

Reply:
Thanks for the details.
I think if you install via `spack install llvm+cuda+omp_debug cuda_arch=70` this should already give you the support for the runtime OpenMP debug options, doesn't it?

Reply:
That's beyond of my level of Spack knowledge, should I check?

Reply:
It should.  If it doesn't that would be good to know.  We could make the decision to always have that turned on, but it isn't a zero cost option at runtime.

Reply:
Can you wait for a bit closing this as I also think `omp_debug` is used to set -DLIBOMPTARGET_ENABLE_DEBUG=ON.
I'll build it on Apollo 70 and share the result here.

Reply:
Hi guys, @trws @naromero77 if we set `omp_debug` to True by default, what's the adverse effect? KMP prints out threw of messages? Spack`llvm`  build becomes like a "Debug" build?

Reply:
This PR should not be needed. It should be handled by the `omp_debug` variant. Perhaps the default should be true since its almost everyone uses it. I am unsure of what this option does in the absence of OpenMP target offload.

Reply:
It builds in the runtime option to get all the debug messages from libomptarget.  It's not a debug build in the sense of compiler flags, but it adds a lot of runtime checks that otherwise wouldn't be there.

Reply:
@trws then you believe `omp_debug` remain False by default?

Reply:
Note that OpenMP offload (`cuda`) is False by default. What's the added value of `omp_debug` being set to True for a OpenMP CPU thread build? (i.e. with no target OpenMP offload).

If there was a simple way to set`omp_debug` to True, when `cuda` is set to True, it would be generally helpful I think.

Reply:
Alright. It is building right now and let you know when I get runtim debug messages from `libomptarget` on ThunderX2.

Reply:
Many thanks to @ax3l , @trws, and @naromero77 , I have built Spack llvm with this:

`spack install llvm+cuda+omp_debug cuda_arch=70` on ThunderX2

and built an OpenMP offloading test app with it. It ran fine (checked offloading part was there with Nsight).